pkg/cmd/dockerregistry/dockerregistry.go

```diff
@@ -98,6 +98,33 @@ func Execute(configFile io.Reader) {
 		pruneAccessRecords,
 	)
 
+	signatureRouter := app.NewRoute().PathPrefix("/signature/").Subrouter()
+	// FIXME: We need more fine-grained role (image-admin).
+	signatureAccessRecords := func(*http.Request) []auth.Access {
+		return []auth.Access{
+			{
+				Resource: auth.Resource{
+					Type: "signature",
+				},
+				Action: "get",
+			},
+		}
+	}
+
+	// Advertise features supported by OpenShift
+	if app.Config.HTTP.Headers == nil {
+		app.Config.HTTP.Headers = http.Header{}
+	}
+	app.Config.HTTP.Headers.Set("X-Registry-Supports-Signatures", "1")
+
+	app.RegisterRoute(
+		// GET /admin/signatures/<reference>
```

Comment:
@miminar I think we don't need to make this repository scoped since you just re-use the token you use for pulling the images. I'm just passing the auth down to Origin API and I'm not doing any complex auth stuff that involve registry service account.